### PR TITLE
align Effect API between scaladsl and javadsl, #25706

### PR DESCRIPTION
**Pull request by patriknw**
scaladsl:
* stop => stop()
* andThenStop() => thenStop()

javadsl:
* andThen => thenRun

Refs #25706

I might want to do some refactoring of the internals of the effects also, but that can be a separate PR.

### CODE REVIEW

File: akka-persistence-typed/src/main/scala/akka/persistence/typed/scaladsl/Effect.scala [hunk 1]
```diff
@@ -58,7 +58,7 @@ object Effect {
    * Stop this persistent actor
    * Side effects can be chained with `andThen`
    */
-  def stop[Event, State]: Effect[Event, State] = none.andThenStop()
+  def stop[Event, State](): Effect[Event, State] = none.thenStop()
```
Comment (patriknw) [inline akka-persistence-typed/src/main/scala/akka/persistence/typed/scaladsl/Effect.scala]: My thinking here is that even though the method is just returning a directive the end result is a side effect and thereby motivates the `()`. Compared to `none` and `unhandled` that don't result in any side effects (well, unhandled might log something)

**Review by chbatey — APPROVED** [review]
LGTM

File: akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/Effect.scala [hunk 2]
```diff
@@ -40,7 +45,7 @@ object EffectFactory extends EffectFactories[Nothing, Nothing, Nothing]
   /**
    * Stop this persistent actor
    */
-  def stop: Effect[Event, State] = none.thenStop()
+  def stop(): Effect[Event, State] = none.thenStop()
```
[the remark this answers is not in the document]
Comment (patriknw) [reply akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/Effect.scala]: ok, I can add it in javadsl, but for scaladsl my thinking was that `none` and `unhandled` don't result in any side effects (well, unhandled might log something).

[the remark this answers is not in the document]
Comment (patriknw) [reply akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/Effect.scala]: done

**Comment by patriknw** [comment]
> Ah I didn't see the motivation at #25812 (comment) before. OK I guess.

I take that as an approval

**Pull request merged**